src/main/host/descriptor/epoll.c

```diff
@@ -725,6 +725,12 @@ static void _epoll_descriptorStatusChanged(Epoll* epoll, const EpollKey* key) {
                 /* this calls unref on the watch if its in the table */
                 g_hash_table_remove(epoll->ready, key);
             }
+
+            /* if it's closed, then remove it from the watching list */
+            if (watch->flags & EWF_CLOSED) {
+                /* unref gets called on the watch when it is removed from these tables */
+                g_hash_table_remove(epoll->watching, &key);
```

Comment:
Do we ever want the watch to exist in the ready table but not the watching table? I'm not sure if that ever happens, but wondering if we need a
```
g_hash_table_remove(epoll->ready, key);
```
here to make sure that if we're not watching it, it will never be in the ready table either? (Or maybe an assert?)

Reply:
If it's closed, the `_epollwatch_isReady(watch)` check above should always return `false`, and then it will run `g_hash_table_remove(epoll->ready, key)`. So it should already be removed from the ready table. But I'll add an assert here to make sure it's removed from the ready table.